### PR TITLE
[NCL-7921] Get previous results for current task

### PR DESCRIPTION
**Pull request by thescouser89**
This commit only contains DTO changes. Implementation commit will happen in a future PR.

### CODE REVIEW

File: model/src/main/java/org/jboss/pnc/rex/model/Task.java [hunk 1]
```diff
@@ -143,6 +143,9 @@ public class Task {
     @Getter(onMethod_ = @ProtoField(number = 14, defaultValue = "false"))
     private Boolean starting;
 
+    @Getter(onMethod_ = @ProtoField(number = 15))
+    private List<String> previousTaskNames = new ArrayList<>();
```
Comment (michalovjan) [inline model/src/main/java/org/jboss/pnc/rex/model/Task.java:147]: I don't think there is a need for this field. There is a field dependencies, which are the direct dependencies.

File: dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java [hunk 2]
```diff
@@ -60,4 +60,6 @@ public class TaskDTO {
     public Set<String> dependants = new HashSet<>();
 
     public Set<String> dependencies = new HashSet<>();
+
+    public List<String> previousTaskNames = new ArrayList<>();
```
Comment (michalovjan) [inline dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java:64]: I don't think there is a need for this field. There is a field dependencies, which are the direct dependencies.

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Class to specify metadata for a Task.
+ */
+@Getter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+public class ConfigurationDTO {
+
+    public boolean provideDependencyResults = false;
```
Comment (michalovjan) [inline dto/src/main/java/org/jboss/pnc/rex/dto/ConfigurationDTO.java:36]: I'd like passResultsOfDependencies.

File: model/src/main/java/org/jboss/pnc/rex/model/Configuration.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * Class to specify metadata for a Task.
+ */
+@Builder
+@AllArgsConstructor(onConstructor_ = {@ProtoFactory})
+@Slf4j
+@Jacksonized
+public class Configuration {
+
+    /**
+     * Specify whether we want to send the direct dependency results in the StartRequest
```
Comment (michalovjan) [inline model/src/main/java/org/jboss/pnc/rex/model/Configuration.java:38]:
```suggestion
     * Specify whether we want to pass results of direct dependencies in the StartRequest and StopRequest
```

File: model/src/main/java/org/jboss/pnc/rex/model/requests/StartRequest.java [hunk 5]
```diff
@@ -34,4 +36,6 @@ public class StartRequest {
     private final String callback;
 
     private final Object payload;
+
+    private final Map<String, Object> taskResults;
```
Comment (michalovjan) [inline model/src/main/java/org/jboss/pnc/rex/model/requests/StartRequest.java:40]: Super good, I've just realized we would probably want to do this also in StopRequest.

File: model/src/main/java/org/jboss/pnc/rex/model/Configuration.java (hunk 4 above)
Comment (michalovjan) [reply model/src/main/java/org/jboss/pnc/rex/model/Configuration.java:38]: a suggestion for clearer wording

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: model/src/main/java/org/jboss/pnc/rex/model/Task.java [hunk 6]
```diff
@@ -143,6 +143,9 @@ public class Task {
     @Getter(onMethod_ = @ProtoField(number = 14, defaultValue = "false"))
     private Boolean starting;
 
+    @Getter(onMethod_ = @ProtoField(number = 15))
+    private Configuration configuration = new Configuration(false);
```
[the remark this answers is not in the document]
Comment (thescouser89) [reply model/src/main/java/org/jboss/pnc/rex/model/Task.java:147]: It's to provide a default value for the Configuration. THe false is that we don't want to get direct dependency results by default.

File: model/src/main/java/org/jboss/pnc/rex/model/requests/StartRequest.java (hunk 5 above)
Comment (thescouser89) [reply model/src/main/java/org/jboss/pnc/rex/model/requests/StartRequest.java:40]: Will add it! Thanks for the suggestion! :)

File: dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java [hunk 7]
```diff
@@ -60,4 +60,6 @@ public class TaskDTO {
     public Set<String> dependants = new HashSet<>();
 
     public Set<String> dependencies = new HashSet<>();
+
+    public ConfigurationDTO configuration = new ConfigurationDTO();
```
Comment (michalovjan) [inline dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java:64]: I've just remembered. You have to add it also into CreateTaskDTO

File: model/src/main/java/org/jboss/pnc/rex/model/Task.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (michalovjan) [reply model/src/main/java/org/jboss/pnc/rex/model/Task.java:147]: I think this is unnecessary. DTO has defaults and Task.class has AllArgsConstructor.

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: model/src/main/java/org/jboss/pnc/rex/model/Task.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (thescouser89) [reply model/src/main/java/org/jboss/pnc/rex/model/Task.java:147]: ok then, i'll remove it.

File: dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java (hunk 7 above)
Comment (thescouser89) [reply dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java:64]: done

**Review by michalovjan — APPROVED** [review]

**Pull request merged**